pkg/network/protocols/http/ebpf_javatls.go

```diff
@@ -78,7 +106,40 @@ func (p *JavaTLSProgram) GetAllUndefinedProbes() (probeList []manager.ProbeIdent
 	return
 }
 
+// isAttachmentAllowed will return true if the pid must not be attach
+// The filter is based on the process command line matching javaAgentAllowRegex and javaAgentBlockRegex regex
+// javaAgentAllowRegex have an higher priority
+func isAttachmentAllowed(pid uint32) bool {
+	// filter is disabled (default configuration)
+	if javaAgentAllowRegex == nil && javaAgentBlockRegex == nil {
+		return true
+	}
+
+	procCmdline := fmt.Sprintf("%s/%d/cmdline", util.HostProc(), pid)
+	cmd, err := os.ReadFile(procCmdline)
+	if err != nil {
+		log.Debugf("injectionFilter can't open comandline %s : %s", procCmdline, err)
+		return false
+	}
+	fullCmdline := strings.ReplaceAll(string(cmd), "\000", " ") // /proc/pid/cmdline format : arguments are separated by '\0'
+
+	// Allow have an higher priority
+	if javaAgentAllowRegex != nil && javaAgentAllowRegex.MatchString(fullCmdline) {
+		return true
+	}
+	if javaAgentBlockRegex != nil && javaAgentBlockRegex.MatchString(fullCmdline) {
+		return false
+	}
+	// by default we attach the java process
+	return true
+}
+
 func newJavaProcess(pid uint32) {
+	if !isAttachmentAllowed(pid) {
+		log.Debugf("java pid %d attachment rejected by blocklist", pid)
```

Comment:
It could be denied if it didn't match the allowlist
```suggestion
		log.Debugf("java pid %d attachment rejected", pid)
```

Reply:
only the blocklist or an error would return false
but "by blocklist" doesn't help here